src/main/java/uk/gov/pay/api/resources/PaymentsResource.java

```diff
@@ -92,7 +93,7 @@ public PaymentsResource(Client client,
                     "as 'authorization: Bearer YOUR_API_KEY_HERE'",
             code = 200)
     @ApiResponses(value = {
-            @ApiResponse(code = 200, message = "OK", response = PaymentWithAllLinks.class),
+            @ApiResponse(code = 200, message = "OK", response = PaymentForSearchResult.class),
```

Comment:
What's the reason for this change? Do the generated structure for the 2 differ? There's a change to PaymentWithAllLinks in this PR to make the docs generate unwrapped, does this make this unnecessary?

Reply:
Yes, the structures differ. PaymentWithAllLinks doesn't include card details as `Payment` object is used here. PaymentForSearchResult inherits from CardPayment which includes all attributes required.

Changes to PaymentWithAllLinks fixes issues with /v1/payments (POST) endpoint .i.e., unwraps payment object for new payment response